src/openrct2/world/Entity.cpp

```diff
@@ -239,7 +239,7 @@ void Vehicle::Serialise(DataSerialiser& stream)
     stream << Pitch;
     stream << bank_rotation;
     stream << remaining_distance;
-    stream << velocity;
+    stream << velocity.getRaw();
```

Comment:
This is broken you can't do this. You need to define a new serialising trait for the Speed16 and Speed32